src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -1098,28 +1253,56 @@ bool is_object_properties_superset(
             auto regex = re2::RE2(as_string_view(propPattern));
             if (re2::RE2::PartialMatch(pname, regex)) {
                 pattern_match_found = true;
-                if (!is_superset(ctx, schemaPattern, schema)) {
-                    // not compatible
-                    return false;
+
+                auto prop_res = is_superset(
+                  ctx, schemaPattern, schema, prop_path);
+
+                if (prop_res.has_error()) {
+                    res.merge(std::move(prop_res));
+                    res.emplace<json_incompatibility>(
+                      std::move(prop_path),
+                      json_incompatibility_type::
+                        property_removed_not_covered_by_partially_open_content_model);
                 }
+                break;
             }
         }
+        if (pattern_match_found) {
+            // check next property
+            continue;
+        }
 
         // or it should check against older["additionalProperties"], if no match
         // in patternProperties was found
-        if (
-          !pattern_match_found
-          && !is_superset(ctx, older_additional_properties, schema)) {
-            // not compatible
-            return false;
+        if (!is_false_schema(older_additional_properties)) {
```

Comment:
This doesn't change the set of schema pairs that are compatible (because a false schema won't match against any schema anyway). But it's necessary to report the specific error of `property_removed_from_closed_content_model` when `older_additional_properties` is false. Let me invert the condition here, that seems simpler.